attestation-service/verifier/src/cca/mod.rs

```diff
@@ -60,6 +61,44 @@ struct CcaEvidence {
     token: Vec<u8>,
 }
 
+#[derive(Error, Debug)]
+pub enum CCAError {
+    #[error("CCA verifier must provide report data field!")]
+    ReportDataMissing,
+    #[error("Deserialize CCA Evidence failed: {0}")]
+    FailedtoDeserializeCCAEvidence(String),
+    #[error("CCA Attestation failed with error: {0}")]
+    CCAAttestationFailed(String),
+    #[error("get the decoding key from the pem public key: {0}")]
+    GetDecodingKey(String),
+    #[error("decrypt the ear with the decoding key: {0}")]
+    DecryptDecodingKey(String),
+    #[error("decode nonce byte from ear: {0}")]
+    DecodeNonceByte(String),
+    #[error("report data is different from that in ear's session nonce")]
+    ReportDataMismatch,
+    #[error("no entry found for CCA_SSD_PLATFORM")]
+    NoEntryFound,
+    #[error("init data hash is different from that in CCA token")]
+    InitDataHashMismatch,
+    #[error("Failed to decode base64: {0}")]
+    FailedtoDecodeBase64(String),
+    #[error("get platform evidence from the cca evidence map")]
+    GetPlatformEvidence,
+    #[error("Serde Json Error")]
+    SerdeJson(#[from] serde_json::Error),
```

Comment:
Could we add some extra information to this kind of error? I mean if this error is raised, the user will only see `Serde Json Error` and he will not know what is serded

Reply:
Adding error types with source as serde::json error and whole context for this.